client/scripts/chat.js

```diff
@@ -1,56 +1,146 @@
 export class Chat {
-  constructor (element) {
+  constructor(element) {
     this._chatContainer = element;
     this._messageInput = this._chatContainer.querySelector('#message');
     this._usersList = this._chatContainer.querySelector('#users');
     this._messagesList = this._chatContainer.querySelector('#messages');
     this._chatForm = this._chatContainer.querySelector('#chat-form');
+    this._emojiContainer = this._chatContainer.querySelector('#emoji-container');
+    this._emojiMap = {
+      ':)': 'smile',
+      ':(': 'sad',
+      '(poop)': 'poop',
+      '(heart eyes)': 'hearteyes',
+      '(thumb up)': 'thumbsup'
+    }
   }
 
-  show () {
-    this._chatContainer.classList.remove('hidden');
+  show() {
+    this._imagesMap = {};
+    this._mapEmojiToImages(this._emojiMap)
+        .then((result) => {
+          result.map((item) => {
+            this._imagesMap[item.symbol] = item.src;
+          });
+          this._fillEmojiContainer(this._imagesMap);
+          this._emojiContainer.addEventListener('click', this._handleEmojiSelect.bind(this));
+          this._chatContainer.classList.remove('hidden');
+        });
   }
 
   onMessage(handleSubmit) {
     this._handleMessageSubmit = handleSubmit;
     this._chatForm.addEventListener('submit', this._chatSubmitHandler.bind(this));
   }
 
-  _chatSubmitHandler (event) {
+  _handleEmojiSelect(event) {
+    const target = event.target;
+    if (target.dataset.symbol) {
+      const symbol = target.dataset.symbol;
+      this._messageInput.value += symbol;
+    }
+  }
+
+  _chatSubmitHandler(event) {
     event.preventDefault();
     this._handleMessageSubmit(this._messageInput.value);
     this._addSentMessage(this._messageInput.value);
     this._messageInput.value = '';
   }
 
-  updateUsers (users) {
+  updateUsers(users) {
     this._usersList.innerHTML = '';
     users.forEach((username) => {
-      this._addListElement(username, [ 'user' ], this._usersList);
+      this._addListElement(username, ['user'], this._usersList);
+    });
+  }
+
+  _fillEmojiContainer(map) {
+    Object.keys(map).forEach((key) => {
+      const image = this._createImage();
+      image.src = map[key];
+      image.classList.add('emoji__select');
+      image.dataset.symbol = key;
+      this._emojiContainer.appendChild(image);
     });
   }
 
   _addSentMessage(msg) {
-    this._addListElement(msg, ['message', 'message-dispatched'], this._messagesList);
+    msg = this._replaceSymbolsWithEmoji(msg, this._imagesMap);
```

Comment:
do not assign to arguments and also try to not reassign variables - it is confusing